treeherder/webapp/api/performance_data.py

```diff
@@ -1,126 +1,104 @@
-        except:
-            return Response("incorrect parameters", 400)
-
-        summary = jm.get_performance_series_summary(interval_seconds, platform)
-        return Response(summary)
-
-    @list_route()
-    @with_jobs
-    def get_signatures_from_properties(self, request, project, jm, pk=None):
-        """
-        GET method implementation for signature data
-
-        Input: property/value pairs
-        Output: unique signatures
-        """
-
-        try:
-            props = request.QUERY_PARAMS.dict()
-        except Exception:
-            return Response("incorrect parameters", 400)
-
-        signatures = jm.get_signatures_from_properties(props)
-
-        return Response(signatures)
-
-    @list_route()
-    @with_jobs
-    def get_signature_properties(self, request, project, jm, pk=None):
-        """
-        GET method for signature
-
-        Input: Signature SHA1's
-        Output: List of signature properties
-        """
-        try:
-            signatures = request.QUERY_PARAMS.getlist("signatures")
-        except Exception:
-            return Response("incorrect parameters", 400)
-
-        if not signatures:
-            return Response("no signatures provided", 400)
-
-        data = jm.get_signature_properties(signatures)
-
-        return Response(data)
-
-    @list_route()
-    @with_jobs
-    def get_performance_data(self, request, project, jm, pk=None):
-        """
-        GET method implementation for performance data
-
-        Input: list of series signatures
-        Output: performance charting data
-        """
-
-        try:
-            signatures = request.QUERY_PARAMS.getlist("signatures")
-            interval = abs(int(request.QUERY_PARAMS.get("interval_seconds", 0)))
-        except Exception:
-            return Response("incorrect parameters", 400)
 
-        if not signatures:
-            return Response("no signatures provided", 400)
-
-        data = jm.get_performance_series_from_signatures(signatures, interval)
-
-        return Response(data)
-
-    @list_route()
-    @with_jobs
-    def platforms(self, request, project, jm, pk=None):
-        """
-        Return list of platforms for which we have performance data
+class PerformanceDatumViewSet(viewsets.ViewSet):
+    """
+    This view serves performance test result data
+    """
+    def list(self, request, project):
+        repository = models.Repository.objects.get(name=project)
 
-        Input: time interval
-        Output: list of platforms
-        """
         try:
-            interval = int(request.QUERY_PARAMS.get('interval'))
+            signature_hashes = request.QUERY_PARAMS.getlist("signatures")
         except:
-            return Response("incorrect parameters", 400)
-
-        platforms = jm.get_performance_platforms(interval)
-
-        return Response(platforms)
+            raise exceptions.ValidationError('need signature list')
+
+        signature_ids = PerformanceSignature.objects.filter(
```

Comment:
Splitting this query in 2 seems a bit too much,. I guess the signature table will never be gigantic and we can always change it later if we see that it doesn't perform well